server/src/main/java/org/opensearch/common/geo/PointParser.java

```diff
@@ -0,0 +1,227 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.common.geo;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.geometry.Point;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Locale;
+
+/**
+ * A utility class with a point parser methods supporting different point representation formats
+ *
+ * @opensearch.internal
+ */
+public class PointParser {
+    private static final String FN_GEOJSON_TYPE = "type";
+    private static final String FV_GEOJSON_TYPE_POINT = "Point";
+    private static final String FN_GEOJSON_COORDS = "coordinates";
+
+    private static final String FN_GEOHASH = "geohash";
+    private static final String ERR_MSG_INVALID_TOKEN = "token [{}] not allowed";
+
+    private String fieldNameX;
+    private String fieldNameY;
+    private boolean supportGeoHash;
+
+    private String invalidFieldErrMsg;
+
+    public PointParser(final String fieldNameX, final String fieldNameY, final boolean supportGeoHash) {
+        this.fieldNameX = fieldNameX;
+        this.fieldNameY = fieldNameY;
+        this.supportGeoHash = supportGeoHash;
+        this.invalidFieldErrMsg = supportGeoHash
+            ? String.format(
+                Locale.ROOT,
+                "field must be either %s/%s, %s/%s, or %s",
+                fieldNameX,
+                fieldNameY,
+                FN_GEOJSON_TYPE,
+                FN_GEOJSON_COORDS,
+                FN_GEOHASH
+            )
+            : String.format(
+                Locale.ROOT,
+                "field must be either %s/%s, or %s/%s",
+                fieldNameX,
+                fieldNameY,
+                FN_GEOJSON_TYPE,
+                FN_GEOJSON_COORDS
+            );
```

Comment:
These error messages can be created at the class who is the caller, rather than we creating it in this class. 

This make PointParser class understand who is the caller and what type of GeoJson it is which is not the responsibility of PointParser class.

Reply:
+1

Reply:
Got the point after discussing with @VijayanB offline. I was having geospatial plugin implementation in mind from the start. I might want to rewrite the entire code.